onnxruntime/core/providers/cuda/cu_inc/variadic_elementwise_impl.cuh

```diff
@@ -57,14 +57,14 @@ __global__ void VariadicElementWiseNoBroadcastInputBatchKernel(
 // assumptions:
 // - inputs.Size() > 1 && inputs.Size() <= max_input_batch_size
 // - inputs and output have N elements
-template <typename T, typename Func, size_t max_input_batch_size>
+template <typename T, typename Func, int32_t max_input_batch_size>
 void VariadicElementWiseNoBroadcastInputBatchImpl(
     Func func,
     size_t N,
     TArray<const T*, max_input_batch_size> inputs,
     T* output) {
-  const size_t elements_per_thread = GridDim::maxElementsPerThread;
-  const size_t threads_per_block = GridDim::maxThreadsPerBlock;
+  constexpr size_t elements_per_thread = GridDim::maxElementsPerThread;
+  constexpr size_t threads_per_block = GridDim::maxThreadsPerBlock;
```

Comment:
change data type of elements_per_thread and threads_per_block to int32 too. In general, only N could be bigger than UINT_MAX.